### ISSUE TITLE
KQLValidations function ValidateKqlForLatestTI is too restrictive

### ISSUE THREAD

**Issue by kfriede** [Azure/Azure-Sentinel#9182]
**Describe the bug**
The ValidateKqlForLatestTI function in KQLValidations contains a regex to validate "the KQL query for the latest Threat Intelligence data. This regex is overly-rigid, and fails when trying to performance-optimize the queries by putting any applicable WHERE filters before the SUMMARIZE. This prevents analysts who contribute analytics/hunting rules from providing working solutions, since the ThreatIntelligenceIndicator dataset can be large and cause the query to consume to many resources. 

**To Reproduce**
1. Write a KQL query using the ThreatIntelligenceIndicator table
2. Include the basic requirements to meet the existing regex (e.g. below)

```
ThreatIntelligenceIndicator
| where TimeGenerated >= ago(ioc_lookBack)
| summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
| where ExpirationDateTime > now() and Active == true
```

3. Add any WHERE filter before the SUMMARIZE (e.g. `| where Active == true`)
4. Run KQLValidations to view failure

**Expected behavior**
KQLValidations passing on any valid KQL prior to SUMMARIZE

**Additional context**
PR #9181 is blocked by this issue

**Comment by v-rbajaj** [comment]
Hi @kfriede, we are reviewing your raised PR for this issue.

**Comment by v-rbajaj** [comment]
Hi @kfriede, the raised PR is still under review.

**Comment by v-rbajaj** [comment]
Hi @kfriede, the raised PR is still under review.

**Comment by v-rbajaj** [comment]
Hi @kfriede, the raised PR is still under review.

**Comment by v-rbajaj** [comment]
Hi @kfriede, the raised PR is still under review.

**Comment by v-muuppugund** [comment]
Hi @kfriede ,We are still working on the PR review internally, will get back to you with updates by 23Nov23

**Comment by v-rbajaj** [comment]
Hi @kfriede, we are still reviewing the PR, will get back to you by 28 Nov 2023.

**Comment by blauwers** [comment]
This issue is resolved in PR #9369. As demonstrated there the `| where Active == true and ExpirationDateTime > now()` need to follow the `| summarize` statement or false positives are introduced.

**Comment by v-rbajaj** [comment]
Hi @blauwers, thanks for providing an update here. We are looking into #9369 PR which is currently under review.
